File: atcoder/ABC/100/C-Mulitply-3-or-Divide-2.rb
n = gets.to_i
a = gets.chomp.split(" ").map(&:to_i)

def div2(x)
    cnt = 0
    while x & 1 == 0
        x = x >> 1
        cnt += 1
    end
    return cnt
end

ans = 0
a.each do |val|
    ans += div2(val)
end
print(ans)

File: atcoder/ABC/100/C-Mulitply-3-or-Divide-2.py
n = int(input())
a = list(map(int, input().split()))

def div2(x):
    cnt = 0;
    while x & 1 == 0:
        x = x >> 1
        cnt += 1
    return cnt

ans = 0;
for x in a:
    ans += div2(x)
print(ans)
